gax-grpc/src/main/java/com/google/api/gax/grpc/GrpcDirectCallable.java

```diff
@@ -44,18 +45,25 @@
  */
 class GrpcDirectCallable<RequestT, ResponseT> extends UnaryCallable<RequestT, ResponseT> {
   private final MethodDescriptor<RequestT, ResponseT> descriptor;
+  private final boolean awaitTrailers;
 
-  GrpcDirectCallable(MethodDescriptor<RequestT, ResponseT> descriptor) {
+  GrpcDirectCallable(MethodDescriptor<RequestT, ResponseT> descriptor, boolean awaitTrailers) {
     this.descriptor = Preconditions.checkNotNull(descriptor);
+    this.awaitTrailers = awaitTrailers;
   }
 
   @Override
   public ApiFuture<ResponseT> futureCall(RequestT request, ApiCallContext inputContext) {
     Preconditions.checkNotNull(request);
     Preconditions.checkNotNull(inputContext);
 
-    return new ListenableFutureToApiFuture<>(
-        ClientCalls.futureUnaryCall(GrpcClientCalls.newCall(descriptor, inputContext), request));
+    ClientCall<RequestT, ResponseT> clientCall = GrpcClientCalls.newCall(descriptor, inputContext);
+
+    if (awaitTrailers) {
+      return new ListenableFutureToApiFuture<>(ClientCalls.futureUnaryCall(clientCall, request));
+    } else {
+      return GrpcClientCalls.eagerFutureUnaryCall(clientCall, request);
```

Comment:
So this is is not wrapped with `ListeneableFutureToApiFuture` why?

Reply:
GrpcClientCalls  is owned by gax, so it already returns an ApiFuture, whereas ClientCalls is owned by grpc returns a ListenableFuture